bin/oneoff/backfill_data/backfill_script_ids_in_teacher_feedbacks_2.rb

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+# Backfill existing TeacherFeedbacks to set script_id based on script_level_id.
+
+require 'optparse'
+
+$options = {actually_update: false}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} [options]"
+  opts.on('-u', '--actually-update', 'Actually perform the update.') do
+    $options[:actually_update] = true
+  end
+  opts.on('-h', '--help', 'Add -u to perform the update.') do
+    puts opts
+    exit
+  end
+end.parse!
+puts "Called with options: #{$options}"
+
+require_relative '../../../dashboard/config/environment'
+
+$stdout.sync = true
+
+def feedbacks_without_script_id
+  TeacherFeedback.where(script_id: nil)
+end
+
+def puts_count
+  puts "There are #{feedbacks_without_script_id.count} feedbacks without a script_id"
+end
+
+def update_script_ids
+  puts "starting backfilling script_ids"
+  feedbacks_without_script_id.find_in_batches(batch_size: 1000) do |batch|
+    print '.'
+    ActiveRecord::Base.transaction do
```

Comment:
Do we want each batch of 1000 records to succeed or fail as a unit? If not, I suggest using a separate transaction for each record within the inner loop.

Reply:
I would say "yes". this rollback should only happen if we did not specify actually_update, in which case we want all DB updates to be rolled back. if any other kind of error occurs, we report it and continue.

Reply:
Ah - the first TeacherFeedback backfill script also worked this way (wrapped each batch of 1000 record updates in a transaction), and somehow I didn't notice it then. Is there any functional reason for batches of 1000 record updates to succeed or fail as a unit? I don't see one. There are also potentially negative database performance impacts to holding a transaction open over 1000 updates. Could we use the [logic from this recent backfill script](https://github.com/code-dot-org/code-dot-org/blob/380d3f5dc53a0712600c34a31ef3d148c6114a22/bin/oneoff/data_fix/fix_invalid_users#L63-L75) that limits the transaction scope to each individual record. Or is that too noisy when it's executed in "dry run" mode because each row is logged as having been rolled back?

Reply:
apologies, Suresh, I miscopied the example you linked to in the first PR. I've updated this one to do the transaction inside the batch. since there are only 100s of rows left to update in production, I am not too worried about noise, so I'm happy to go with your recommendation here.